### PR TITLE
Refactor AWS Integration tests

### LINKED ISSUE #19052: Refactor Wazuh core repository AWS integration tests
| Affected integration |
|---|
| AWS|

## Description

As part of epic https://github.com/wazuh/wazuh/issues/17788, AWS ITs should be updated and refactored to adapt to guidelines described in epic and more specifically to the latest changes in the AWS module of the framework.

Some of the changes to be made if possible are:

- [x] Fix all expected import errors due to the latest framework changes.
- [x] Modularization of repeated configurations.  
- [x] Analyze current use cases to check for updates.
- [x] Clean code.

### PR DESCRIPTION
**Pull request by EduLeon12**
|Related issue|
|---|
| Closes #19052 |

## Description

As part of epic #17788, Issue tends to refactor de AWS ITs for better readability, modularization, and implementation of the migrated qa-integration-framework.

## Tests

Test summary can be seen [here](https://github.com/wazuh/wazuh/issues/19052#issuecomment-1821386027) and the entire suite summary [here](https://github.com/wazuh/wazuh/issues/19052#issuecomment-1828688537).

### CODE REVIEW

File: tests/integration/test_aws/test_only_logs_after.py [hunk 1]
```diff
@@ -1,53 +1,47 @@
...
+# Set test configurator for the module
+configurator = TestConfigurator(module='only_logs_after_test_module')
 
-# Generic vars
-MODULE = 'only_logs_after_test_module'
-TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
-CONFIGURATIONS_PATH = os.path.join(TEST_DATA_PATH, TEMPLATE_DIR, MODULE)
-TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, TEST_CASES_DIR, MODULE)
+import pydevd_pycharm
+pydevd_pycharm.settrace('192.168.56.1', port=55555, stdoutToServer=True, stderrToServer=True)
```
Comment (nico-stefani) [inline tests/integration/test_aws/test_only_logs_after.py:33]: These lines must be removed

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: tests/integration/test_aws/test_only_logs_after.py (hunk 1 above)
Comment (EduLeon12) [reply tests/integration/test_aws/test_only_logs_after.py:33]: Done

File: tests/integration/test_aws/conftest.py [hunk 2]
```diff
@@ -14,8 +25,21 @@
...
+)
+from wazuh_testing.modules.monitord import configuration as monitord_config
+
+# Local imports
+from .utils import TEST_DATA_PATH,  TEMPLATE_DIR, TEST_CASES_DIR, WAZUH_MODULES_DEBUG
+
+
+# Set local internal options
+local_internal_options = {WAZUH_MODULES_DEBUG: '2',
+                          monitord_config.MONITORD_ROTATE_LOG: '0'}
```
[the remark this answers is not in the document]
Comment (EduLeon12) [reply tests/integration/test_aws/conftest.py:42]: Done

File: tests/integration/test_aws/test_only_logs_after.py [hunk 3]
```diff
@@ -925,39 +937,55 @@ def test_cloudwatch_multiple_calls(
         '--debug', '2'
     ]
 
+    if service_type == INSPECTOR_TYPE:
+        pattern = fr"{NO_NEW_EVENTS}"
+    else:
+        pattern = fr"{EVENT_SENT}"
```
[the remark this answers is not in the document]
Comment (EduLeon12) [reply tests/integration/test_aws/test_only_logs_after.py]: Removed !

**Comment by EduLeon12** [comment]
After the changes tests were rerunned with the same result as before.

```python
root@vagrant:/wazuh/tests/integration# pytest --disable-warnings test_aws/
=========================== test session starts ===========================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.3.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.0.0, html-3.1.1
collected 200 items

test_aws/test_basic.py ................                                                                                      [  8%]
test_aws/test_custom_bucket.py ..                                                                                            [  9%]
test_aws/test_discard_regex.py .................                                                                             [ 17%]
test_aws/test_log_groups.py ..                                                                                               [ 18%]
test_aws/test_only_logs_after.py .................................. [ 35%]
...........x.                                                                       [ 42%]
test_aws/test_parser.py ..............F...........                                  [ 55%]
test_aws/test_path.py ..........................................                    [ 76%]
test_aws/test_path_suffix.py .........                                              [ 80%]
test_aws/test_regions.py ........................                                   [ 92%]
test_aws/test_remove_from_bucket.py ...sss.........                                 [100%]
====== 1 failed, 195 passed, 3 skipped, 1 xfailed, 18 warnings in 4043.42s (1:07:23) ======

```

**Review by nico-stefani — APPROVED** [review]
LGTM

File: tests/integration/test_aws/conftest.py [hunk 4]
```diff
@@ -1,4 +1,14 @@
+# Copyright (C) 2015-2023, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+"""
+This module contain all necessary components (fixtures, classes, methods)to configure the test for its execution.
```
Comment (Selutario) [inline tests/integration/test_aws/conftest.py:6]:
```suggestion
This module contains all necessary components (fixtures, classes, methods) to configure the test for its execution.
```

File: tests/integration/test_aws/test_parser.py [hunk 5]
```diff
@@ -447,32 +423,31 @@ def test_invalid_values_in_bucket(
     input_description:
         - The `configuration_values_in_bucket` file provides the configuration for this test.
     """
+
+
+
```
Comment (Selutario) [inline tests/integration/test_aws/test_parser.py:428]:
```suggestion
```

File: tests/integration/test_aws/test_only_logs_after.py [hunk 6]
```diff
@@ -285,21 +273,19 @@ def test_service_without_only_logs_after(
         callback=event_monitor.callback_detect_all_aws_err
     )
 
-    assert log_monitor.callback_result is None, ERROR_MESSAGES['error_found']
+    assert log_monitor.callback_result is None, ERROR_MESSAGE['error_found']
```
Comment (Selutario) [inline tests/integration/test_aws/test_only_logs_after.py:276]: `expected_results = metadata['expected_results']` is being declared inside this test but is never used, is that correct?

File: tests/integration/test_aws/utils.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+                                       "did not process the expected amount of logs",
+    "failed_sqs_message_retrieval": "The AWS module did not retrieve the expected message from the SQS Queue",
+    "failed_message_handling": "The AWS module did not handle the expected message"
+}
+
+TIMEOUT = {
+
+    10: 10,
+    20: 20
+}
```
Comment (Selutario) [inline tests/integration/test_aws/utils.py:53]:
```suggestion
TIMEOUT = {
    10: 10,
    20: 20
}
```

**Review by Selutario — CHANGES_REQUESTED** [review]
Looks good, just some minor details:

File: tests/integration/test_aws/test_only_logs_after.py (hunk 6 above)
Comment (EduLeon12) [reply tests/integration/test_aws/test_only_logs_after.py:276]: No is not, it's unused coded and has been removed.

**Comment by EduLeon12** [comment]
@Selutario All comments have been addressed, and the necessary changes have been implemented. Since these modifications primarily involved formatting adjustments, integration tests were not re-launched, except for the one where the unused variable was removed.

``` python

root@vagrant:/wazuh/tests/integration/test_aws# pytest test_only_logs_after.py::test_service_without_only_logs_after
=================================================== test session starts ====================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.3.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.0.0, html-3.1.1
collected 1 item

test_only_logs_after.py .                                                                                            [100%]

```

**Review by Selutario — APPROVED** [review]

**Pull request merged**